### ISSUE TITLE
TFLiteTransferConverter (tfltransfer) for model personalization cannot convert a model on 'keras_model_head.py': AttributeError: 'list' object has no attribute 'values' line 51

### ISSUE THREAD

**Issue by amrmp** [tensorflow/tensorflow#45949]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): NO
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.4.0
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.1
- GPU model and memory: 750 ti 2GB

**Describe the current behavior**
I am following the model personalization post available on the TensorFlow GitHub repository (https://github.com/tensorflow/examples/blob/master/lite/examples/model_personalization/README.md) to build a custom trainable model on a mobile device. I expect to have the custom (or even the provided example) model for the 'head' can be converted via the following code:
`converter = TFLiteTransferConverter(4,
                                    base,
                                    heads.KerasModelHead(head),
                                    optimizers.SGD(3e-2),
                                    train_batch_size=20)`

But the library gets stuck on `heads.KerasModelHead(head)` line and produces the following error:

```
/tfltransfer/heads/keras_model_head.py in __init__(self, keras_model)
     49     self._predict_signature = loaded_model.signatures['serving_default']
     50 
---> 51     input_def = next(self._predict_signature.inputs.values().__iter__())
     52     self._input_shape = tuple(
     53         dim.size for dim in input_def.tensor_shape.dim[1:])

AttributeError: 'list' object has no attribute 'values'
```

I have tried both the provided example and a custom model on different systems including Google Colab. The issue cannot be resolved.

**Describe the expected behavior**

The converter should produce the converted model files to be used on the device. The converter should produce the following file based on a custom base and head models.
```
bottleneck.tflite
inference.tflite
initialize.tflite
optimizer.tflite
train_head.tflite
```

**Standalone code to reproduce the issue**
The following is the example code obtained from the TensorFlow Github tested on Google Colab.
```
import tensorflow as tf
from google.colab import drive
drive.mount('/content/drive')
nb_path = '/content/drive/MyDrive/ML/converter/'
sys.path.insert(0,nb_path)
from tfltransfer import bases
from tensorflow.keras import layers
from tensorflow.keras.regularizers import l2
from tfltransfer import bases
from tfltransfer import heads
from tfltransfer import optimizers
from tfltransfer.tflite_transfer_converter import TFLiteTransferConverter
base = bases.MobileNetV2Base(image_size=224)

head = tf.keras.Sequential([
layers.Flatten(input_shape=(7, 7, 1280)),
layers.Dense(
units=32,
activation='relu',
kernel_regularizer=l2(0.01),
bias_regularizer=l2(0.01)),
layers.Dense(
units=4,
activation='softmax',
kernel_regularizer=l2(0.01),
bias_regularizer=l2(0.01)),
])
head.compile(loss='categorical_crossentropy', optimizer='sgd')
converter = TFLiteTransferConverter(4,
base,
heads.KerasModelHead(head),
optimizers.SGD(3e-2),
train_batch_size=20)
converter.convert_and_save('custom_keras_model')

```

**Other info / logs** Include any logs or source code that would be helpful to
```
tfltransfer/heads/keras_model_head.py", line 51, in __init__
    input_def = next(self._predict_signature.inputs.values().__iter__())
AttributeError: 'list' object has no attribute 'values'
```

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3, TF v2.4 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/58091b60823af37b3eaaa5c5cb83168c/45949.ipynb). Thanks!

**Comment by jaeyoo** [comment]
Yes, I could reproduce it too. It was due to the recent removing experimental.export_saved_model(). Please let me work on it.

**Comment by jaeyoo** [comment]
It is due to the latest keras API update (e.g. removed experimental saved model function), so I am working on huge internal upgrade from TFv1 to TFv2 in my model personalization library. Please stay tuned.

**Comment by shrutiquoori** [comment]
Is there any update as to the timeline of the upgrade?

**Comment by yefengjie** [comment]
some issue here,@jaeyoo
is there any useful old tensorflow version or keras version?

**Comment by yefengjie** [comment]
> It is due to the latest keras API update (e.g. removed experimental saved model function), so I am working on huge internal upgrade from TFv1 to TFv2 in my model personalization library. Please stay tuned.

some issue here,@jaeyoo
is there any useful old tensorflow version or keras version?

**Comment by Junhyuk** [comment]
@jaeyoo 
Hello 
is here any update?

**Comment by jaeyoo** [comment]
For temporary workaround, please visit this colab:

https://colab.research.google.com/drive/1OFDCHvgvgZkiXjFi_wwCof0O1VWzrsez?resourcekey=0-K6oEwOCLVLH1nao8y7pQgw&usp=sharing

**Comment by sushreebarsa** [comment]
Was able to reproduce the issue in TF v2.5,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/c87052933d0434eecdce861fd2683241/untitled189.ipynb)..Thanks !

**Comment by raggyie** [comment]
@jaeyoo 
Hi any update to the fix still?
Was able to reproduce the issue with tfv2.7.0-dev20210919

**Comment by pjpratik** [comment]
@amrmp The issue should have been fixed with [#1001](https://github.com/adap/flower/pull/1001). Can you check if the issue still persists. Thank you.

**Comment by Junhyuk** [comment]
@pjpratik 
in my case , it might be helpful to try try with tensorflow signature above TF 2.7.
